### ISSUE TITLE
how to use dcnv1 only

### ISSUE THREAD

**Issue by b762927** [developer0hye/Simple-PyTorch-Deformable-Convolution-v2#12]
I just want to use dcn version1 ，how to do th
is？

**Comment by developer0hye** [comment]
@b762927

Refer to the below code.


```python
import torch
import torchvision.ops
from torch import nn

class DeformableConv2d(nn.Module):
    def __init__(self,
                 in_channels,
                 out_channels,
                 kernel_size=3,
                 stride=1,
                 padding=1,
                 bias=False):

        super(DeformableConv2d, self).__init__()
        
        assert type(kernel_size) == tuple or type(kernel_size) == int

        kernel_size = kernel_size if type(kernel_size) == tuple else (kernel_size, kernel_size)
        self.stride = stride if type(stride) == tuple else (stride, stride)
        self.padding = padding
        
        self.offset_conv = nn.Conv2d(in_channels, 
                                     2 * kernel_size[0] * kernel_size[1],
                                     kernel_size=kernel_size, 
                                     stride=stride,
                                     padding=self.padding, 
                                     bias=True)

        nn.init.constant_(self.offset_conv.weight, 0.)
        nn.init.constant_(self.offset_conv.bias, 0.)
        
        self.regular_conv = nn.Conv2d(in_channels=in_channels,
                                      out_channels=out_channels,
                                      kernel_size=kernel_size,
                                      stride=stride,
                                      padding=self.padding,
                                      bias=bias)

    def forward(self, x):
        #h, w = x.shape[2:]
        #max_offset = max(h, w)/4.

        offset = self.offset_conv(x)#.clamp(-max_offset, max_offset)
        
        x = torchvision.ops.deform_conv2d(input=x, 
                                          offset=offset, 
                                          weight=self.regular_conv.weight, 
                                          bias=self.regular_conv.bias, 
                                          padding=self.padding,
                                          stride=self.stride,
                                          )
        return x
```

**Comment by b762927** [comment]
Thank you very much. You are a warm-hearted person! Thank you again for your help!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "developer0hye/PyTorch-Deformable-Convolution-v2"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月1日(星期三) 晚上9:35
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [developer0hye/PyTorch-Deformable-Convolution-v2] how to use dcnv1 only (Issue #12)





 import torch import torchvision.ops from torch import nn class DeformableConv2d(nn.Module):     def __init__(self,                  in_channels,                  out_channels,                  kernel_size=3,                  stride=1,                  padding=1,                  bias=False):         super(DeformableConv2d, self).__init__()                  assert type(kernel_size) == tuple or type(kernel_size) == int         kernel_size = kernel_size if type(kernel_size) == tuple else (kernel_size, kernel_size)         self.stride = stride if type(stride) == tuple else (stride, stride)         self.padding = padding                  self.offset_conv = nn.Conv2d(in_channels,                                       2 * kernel_size[0] * kernel_size[1],                                      kernel_size=kernel_size,                                       stride=stride,                                      padding=self.padding,                                       bias=True)         nn.init.constant_(self.offset_conv.weight, 0.)         nn.init.constant_(self.offset_conv.bias, 0.)                  self.regular_conv = nn.Conv2d(in_channels=in_channels,                                       out_channels=out_channels,                                       kernel_size=kernel_size,                                       stride=stride,                                       padding=self.padding,                                       bias=bias)     def forward(self, x):         #h, w = x.shape[2:]         #max_offset = max(h, w)/4.         offset = self.offset_conv(x)#.clamp(-max_offset, max_offset)                  x = torchvision.ops.deform_conv2d(input=x,                                            offset=offset,                                            weight=self.regular_conv.weight,                                            bias=self.regular_conv.bias,                                            padding=self.padding,                                           stride=self.stride,                                           )         return x
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by developer0hye** [comment]
@b762927 
Thanks~
